pkg/controller/core/capacity_controller.go

```diff
@@ -47,47 +55,121 @@ func NewCapacityReconciler(cache *capacity.Cache) *CapacityReconciler {
 //+kubebuilder:rbac:groups=kueue.x-k8s.io,resources=capacities/finalizers,verbs=update
 
 func (r *CapacityReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	// No-op. All work is done in the event handler.
+	var capObj kueue.Capacity
+	if err := r.client.Get(ctx, req.NamespacedName, &capObj); err != nil {
+		// we'll ignore not-found errors, since there is nothing to do.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	log := ctrl.LoggerFrom(ctx).WithValues("capacity", klog.KObj(&capObj))
+	ctx = ctrl.LoggerInto(ctx, log)
+	log.V(2).Info("Reconciling Capacity")
+
+	usage, workloads, err := r.cache.Usage(&capObj)
+	if err != nil {
+		log.Error(err, "Failed getting usage from capacity cache")
+		// This is likely because the capacity was recently removed and we didn't
+		// process that event yet.
+		return ctrl.Result{}, err
```

Comment:
Maybe in this scenario we want to requeue? 
```go
	// Requeue tells the Controller to requeue the reconcile key.  Defaults to false.
	Requeue bool
```
to trigger a new try to reconcile the state of the capacity?